src/builder/QuarkBuilder.sol

```diff
@@ -261,28 +262,53 @@ contract QuarkBuilder {
     }
 
     // Assert that each chain has sufficient funds to cover the max cost for that chain.
-    // NOTE: This check assumes we will not be bridging payment tokens for the user.
-    function assertPaymentAffordable(
-        TransferIntent memory transferIntent,
-        Accounts.ChainAccounts[] memory chainAccountsList,
-        Payment memory payment
-    ) internal pure {
-        if (payment.isToken) {
-            for (uint256 i = 0; i < payment.maxCosts.length; ++i) {
-                uint256 paymentAssetBalanceOnChain = Accounts.sumBalances(
-                    Accounts.findAssetPositions(payment.currency, payment.maxCosts[i].chainId, chainAccountsList)
-                );
-                uint256 paymentAssetNeeded = payment.maxCosts[i].amount;
+    // Check user account can cover the cost of each actions
+    function assertActionsAffordable(Actions.Action[] memory actions, Accounts.ChainAccounts[] memory chainAccountsList)
```

Comment:
Now the affordable is check actions can be covered by the payment maxCost at the end of the build.

Reply:
Nice, makes sense to me 👍